plugins/module_utils/network/sonic/config/bgp_neighbors/bgp_neighbors.py

```diff
@@ -420,13 +428,44 @@ def build_bgp_peer_groups_payload(self, cmd, have, bgp_as, vrf_name):
                     if peer_group['address_family'].get('afis', None) is not None:
                         for each in peer_group['address_family']['afis']:
                             samp = {}
+                            afi_safi_cfg = {}
+                            pfx_lmt_cfg = {}
+                            pfx_lst_cfg = {}
+                            ip_dict = {}
                             if each.get('afi', None) is not None and each.get('safi', None) is not None:
                                 afi_safi = each['afi'].upper() + "_" + each['safi'].upper()
                                 if afi_safi is not None:
                                     afi_safi_name = 'openconfig-bgp-types:' + afi_safi
                                 if afi_safi_name is not None:
                                     samp.update({'afi-safi-name': afi_safi_name})
                                     samp.update({'config': {'afi-safi-name': afi_safi_name}})
+                                if afi_safi == 'IPV4_UNICAST':
```

Comment:
The following blocks of code could be consolidated to eliminate repeated instructions.

e.g.  (pseudo code), but feel free to depart from this in any way that reduces the repetition:

----------

<store pfx_lmt_cfg>
if afi_safi == 'L2VPN_EVPN':
  <conditional samp.update>
else:
  <common ip handling>
  if ip_dict and if afi_safi == 'IPV4_UNICAST':
    samp.update({'ipv4-unicast': ip_dict})
  if ip_dict and if afi_safi == 'IPV6_UNICAST':
    samp.update({'ipv6-unicast': ip_dict})